app/src/main/java/com/example/sharingang/UserProfileFragment.kt

```diff
@@ -1,51 +1,179 @@
 package com.example.sharingang
 
+
+import android.net.Uri
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Button
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
 import androidx.navigation.fragment.navArgs
 import com.bumptech.glide.Glide
 import com.example.sharingang.databinding.UserProfileFragmentBinding
 import com.example.sharingang.users.CurrentUserProvider
+import com.example.sharingang.users.User
+import com.example.sharingang.users.UserRepository
+import com.example.sharingang.utils.ImageAccess
 import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
+
+
 @AndroidEntryPoint
 class UserProfileFragment : Fragment() {
     private val userViewModel: UserProfileViewModel by viewModels()
     private val args: UserProfileFragmentArgs by navArgs()
     private lateinit var binding: UserProfileFragmentBinding
+    // This is the currently logged in user
+    private var currentUserId: String? = ""
```

Comment:
Do we need "String?" if it is initialized to a non-null value?

Reply:
Maybe just init to null so when we check that a user is logged in it actually triggers the null checks